File: app/src/main/java/com/happs/ximand/ringcontrol/model/object/command/SendDateCommand.java
package com.happs.ximand.ringcontrol.model.object.command;

public class SendDateCommand extends BluetoothCommand<String> {

    private static final byte SEND_TIME_COMMAND_CODE = 30;

    private final byte yearLastTwoDigits;
    private final byte month;
    private final byte dateOfMonth;
    private final byte hours;
    private final byte minutes;
    private final byte seconds;
    private final byte dateOfWeek;

    public SendDateCommand(byte yearLastTwoDigits, byte month, byte dateOfMonth,
                           byte hours, byte minutes, byte seconds, byte dateOfWeek) {
        super(SEND_TIME_COMMAND_CODE);
        this.yearLastTwoDigits = yearLastTwoDigits;
        this.month = month;
        this.dateOfMonth = dateOfMonth;
        this.hours = hours;
        this.minutes = minutes;
        this.seconds = seconds;
        this.dateOfWeek = dateOfWeek;
    }

    @Override
    public byte[] getCommand() {
        return new byte[]{
                getCode(), yearLastTwoDigits, month, dateOfMonth, hours, minutes, seconds, dateOfWeek
        };
    }

    @Override
    public String getMainContent() {
        return hours + ":" + minutes + ":" + seconds;
    }
}

File: app/src/main/java/com/happs/ximand/ringcontrol/model/object/command/SendDateCommand.kt
package com.happs.ximand.ringcontrol.model.`object`.command

import java.util.Calendar.*

class SendDateCommand : BluetoothCommand<String?>(CODE) {

    private val yearLastTwoDigits: Byte
    private val month: Byte
    private val dateOfMonth: Byte
    private val hours: Byte
    private val minutes: Byte
    private val seconds: Byte
    private val dateOfWeek: Byte

    companion object {
        private const val CODE: Byte = 30
    }

    init {
        val calendar = getInstance()
        yearLastTwoDigits = (calendar[YEAR] % 100).toByte()
        month = calendar[MONTH].toByte()
        dateOfMonth = calendar[DAY_OF_MONTH].toByte()
        hours = calendar[HOUR_OF_DAY].toByte()
        minutes = calendar[MINUTE].toByte()
        seconds = calendar[SECOND].toByte()
        dateOfWeek = calendar[DAY_OF_WEEK].toByte()
    }

    override fun toByteArray(): ByteArray {
        return byteArrayOf(
                code, yearLastTwoDigits, month, dateOfMonth, hours, minutes, seconds, dateOfWeek
        )
    }

    override fun getMainContent(): String {
        return "$hours:$minutes:$seconds"
    }

}